hosting/src/hooks/useTanamDocuments.tsx

```diff
@@ -111,6 +121,41 @@ export function useCrudTanamDocument(documentId?: string) {
       setIsLoading(false);
     }
   }
-
   return {update, isLoading, error};
 }
+
+export function useCreateTanamDocument(documentType?: string) {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<UserNotification | null>(null);
+
+  async function create() {
+    setIsLoading(true);
+    try {
+      if (!documentType) {
+        setError(new UserNotification("error", "Missing parameter", "Document id parameter is missing"));
+        return;
+      }
+      const typeRef = collection(firestore, "tanam-documents");
+      const tanamDocument = new TanamDocumentClient("", {
+        data: {content: ""},
+        documentType,
+        revision: 0,
```

Comment:
Do not initialize objects with default values that assume knowledge or doesn't make sense.

# 1
By setting the revision to `0` you are hardcoding the knowledge of revisions into this particular part of the app. If another developer think it starts with `1` you get another functionality in another part of the app.

In this case, you can see it in the class implementation (always know how classes are implemented to behave)
https://github.com/oddbit/tanam/blob/ddebb2ec3e6a8c3f112628946320c99d2fe91b9b/functions/src/models/TanamDocument.ts#L22

Never hardcode default values. Always set them in the constructor or factory creation of classes.

# 2
Never set illegal values to fields such as empty string `""` to the ID. You are creating a dangerous state of the application below and if you make that a habit you will soon find yourself initializing objects with hardcoded IDs or empty IDs that can cause all sorts of unexpected errors. Always make sure that values you set make sense at all times or that the class implementation has a way to handle it.


# Change this to 
```typescript
      const docRef = doc(firestore, "tanam-documents");
      const docId = docRef.id;
      const tanamDocument = new TanamDocumentClient(docId, {
        data: {},
        documentType,
      }).toJson();

      await setDoc(docRef, tanamDocument);
      return docId;
```

Reply:
thanks for the feedback and I'll try to improve it next time 😄